pkgs/cli/src/node.zig

```diff
@@ -815,3 +817,40 @@ test "ENR construction from fields" {
     try std.testing.expect(constructed_enr.kvs.get("tcp") != null);
     try std.testing.expect(constructed_enr.kvs.get("seq") != null);
 }
+
+test "compare roots from genGensisBlock andgenGenesisState and genStateBlockHeader" {
+    var arena_allocator = std.heap.ArenaAllocator.init(std.testing.allocator);
+    defer arena_allocator.deinit();
+    const allocator = arena_allocator.allocator();
+
+    // Load config.yaml from test fixtures
+    const config_filepath = "pkgs/cli/test/fixtures/config.yaml";
+    var parsed_config = try utils.loadFromYAMLFile(allocator, config_filepath);
+    defer parsed_config.deinit(allocator);
+
+    // Parse genesis config from YAML
+    const genesis_spec = try configs.genesisConfigFromYAML(allocator, parsed_config, null);
+    defer allocator.free(genesis_spec.validator_pubkeys);
+
+    // Generate genesis state
+    var genesis_state: types.BeamState = undefined;
+    try genesis_state.genGenesisState(allocator, genesis_spec);
+    defer genesis_state.deinit();
+
+    // Generate genesis block using genGenesisBlock
+    var genesis_block: types.BeamBlock = undefined;
+    try genesis_state.genGenesisBlock(allocator, &genesis_block);
+    defer genesis_block.deinit();
+
+    // Get state root by hashing the state directly
+    var state_root_from_genesis: [32]u8 = undefined;
+    try ssz.hashTreeRoot(types.BeamState, genesis_state, &state_root_from_genesis, allocator);
+
+    // Generate block header using genStateBlockHeader
+    const state_block_header = try genesis_state.genStateBlockHeader(allocator);
```

Comment:
can you hard compare aganist actual roots which for e.g. ream can also verify using the same config.yaml

Reply:
<img width="1493" height="153" alt="image" src="https://github.com/user-attachments/assets/f19932a5-d73d-4b36-87ed-3b1fd50bfbe9" />